otx/cli/builder/builder.py

```diff
@@ -0,0 +1,276 @@
+                f"\n[otx build] Please refer to {inspect.getfile(backbone_function)}"
+            )
+        for arg in missing_args:
+            if arg in required_option:
+                backbone_config[arg] = f"!!!SELECT_OPTION: {required_option[arg]}"
+            else:
+                backbone_config[arg] = "!!!!!!!!!!!INPUT_HERE!!!!!!!!!!!"
+        print(
+            f"[otx build] {backbone_config['type']} requires the argument : {missing_args}"
+            f"\n[otx build] Please refer to {inspect.getfile(backbone_function)}"
+        )
+    backbone_config["use_out_indices"] = use_out_indices
+    return missing_args
+
+
+def update_in_channel(model_config, out_channels):
+    """Update in_channel of head or neck."""
+    if hasattr(model_config.model, "neck"):
+        print(f"\tUpdate model.neck.in_channels: {out_channels}")
+        model_config.model.neck.in_channels = out_channels
+    elif hasattr(model_config.model, "bbox_head"):
+        raise NotImplementedError("This architecture currently does not support public backbone.")
+    elif hasattr(model_config.model, "decode_head"):
+        print(f"\tUpdate model.decode_head.in_channels: {out_channels}")
+        model_config.model.decode_head.in_channels = out_channels
+    elif hasattr(model_config.model, "head"):
+        print(f"\tUpdate model.head.in_channels: {out_channels}")
+        model_config.model.head.in_channels = out_channels
+
+
+class Builder:
+    """Class that implements a model templates registry."""
+
+    def build_task_config(self, task_type, model_type=None, workspace_path=None):
+        """Create OTX workspace with Template configs from task type.
+
+        task_type: The type of task want to get (str)
+        model_type: Specifies the template of a model (str)
+        workspace_path: This is the folder path of the workspace want to create (os.path)
+        """
+
+        # Create OTX-workspace
+        if workspace_path is None:
+            workspace_path = f"./otx-workspace-{task_type}"
+        os.makedirs(workspace_path, exist_ok=True)
+
+        # Load & Save Model Template
+        otx_registry = OTXRegistry("otx").filter(task_type=task_type)
+        if model_type:
+            template = [temp for temp in otx_registry.templates if temp.name.lower() == model_type.lower()]
+            if len(template) == 0:
+                raise ValueError(
+                    f"[otx build] {model_type} is not a type supported by OTX {task_type}."
+                    f"\n[otx build] Please refer to 'otx find --template --task_type {task_type}'"
+                )
+            template = template[0]
+        else:
+            template = otx_registry.get(DEFAULT_MODEL_TEMPLATE_ID[task_type])
+        template_dir = os.path.dirname(template.model_template_path)
+
+        # Copy task configuration file
+        task_configuration_path = os.path.join("/".join(template_dir.split("/")[:-1]), "configuration.yaml")
```

Comment:
Is it safe root for semisl configs? I think it's better to use template.hyper_parameters.base_path for finding configuration.yaml

Reply:
Edited as per your suggestion.
`task_configuration_path = os.path.join(template_dir, template.hyper_parameters.base_path)`